.github/workflows/upstream_merge.yml

```diff
@@ -0,0 +1,38 @@
+name: Merge
+
+on:
+  workflow_dispatch:
+    inputs:
+      Upstream:
+        description: 'Upstream to pull'
+        required: true
+        default: 'NebulaSS13/Nebula'
+      Branch:
+        description: 'Branch of upstrem to pull'
+        required: true
+        default: dev
+
+jobs:
+  Merge:
+    runs-on: ubuntu-latest
+    if: github.repository != ${{ github.event.inputs.Upstream }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+        with:
+          fetch-depth: 25
+      - name: Fetching
+        run: |
+          git remote add upstream https://github.com/${{ github.event.inputs.Upstream }}
+          git fetch --all
+      - name: Merge
+        run: |
+          git config --local user.email "action@github.com"
+          git config --local user.name "Upstream Merger"
+          git merge upstream/${{ github.event.inputs.Branch }}
+      - name: Push
+        uses: ad-m/github-push-action@master
```

Comment:
Isn't the convention to use a specified release version, and then rely on Dependabot or the like to handle (most of) the update process?

Reply:
What?

Reply:
Sooo?

Reply:
One head over to https://github.com/ad-m/github-push-action
One find the releases page https://github.com/ad-m/github-push-action/releases
Pick the latest relevant release and use that. In this case: ad-m/github-push-action@v0.6.0

This avoids issues with hostile and/or breaking changes when/if the workflow is updated

Then one setup a Dependabot. This bot automatically opens PRs when new releases of referenced workflows are released.